### ISSUE TITLE
How i can pass file to other server 

### ISSUE THREAD

**Issue by rustam87** [form-data/form-data#432]
I have express server. 
I get file from browser in my express server
How add this file to form-data that send file to another server?

express

```
const FormData = require('form-data')

in route

const files = req.files (files from browser)
const form = new FormData()

files.forEach((file)=>{
        // in file i have only buffer info
        // if i send file not file.buffer i get error: **source.on is not a function**.
        // therefore i send file.buffer
        form.append('file', file.buffer, file.originalname)
    });

axios.post('anotherserver', form, {headers: 'multipart/form-data'})
```
It right way or not? How i can see my file in formData?
After request to another server i can see 
FormData {
 _overheadLength: 154,
_valueLength: 4096,
_valuesToMeasure: [],
writable: false,
readable: true,
dataSize: 0,
maxDataSize: 2097152,
pauseStreams: true,
_released: true,
_streams: [],
_currentStream: null,
_boundary: '--------------------------181997849912733363348082',
_events: {},
_eventsCount: 0 } },

Where is storage my file?

**Comment by alexindigo** [comment]
FormData is a stream. You can see it in the code, it invokes CombinedStream `append` with the passed value – https://github.com/form-data/form-data/blob/master/lib/form_data.js#L74
Which adds it to the _streams array. And when stream is being read, _streams array is drained.

Try to use `form.submit` from examples, to check if it's issue with axios or not.

**Comment by rustam87** [comment]
I didn't understand 
form.append('file', file.buffer, file.originalname)

Do i need transform file.buffer to stream?
If Yes? how can i do this without or with createReadStream

**Comment by alexindigo** [comment]
No, you don't have to convert buffer to stream for FormData to use it. But in order to access data "in the stream" after it was drained (sent) you need to do extra steps.

**Comment by rustam87** [comment]
> No, you don't have to convert buffer to stream for FormData to use it. But in order to access data "in the stream" after it was drained (sent) you need to do extra steps.

 Can you give me an example? If it is not difficult for you, of course.

**Comment by stormmuller** [comment]
This is also something I'm struggling with. It's a common pattern to pass files along to some storage service. Would be cool if someone who managed to do this form-data and express to push an example to the repo.

**Comment by bradrisse** [comment]
Here is what I used
```
sendToExternalService(file) {
    let self = this
    return new Promise(function(resolve, reject) {
      let formData = new FormData()
      formData.append("image", file.buffer, file.originalname)
      formData.submit(
        {
          protocol: "https:",
          host: `example.com`,
          path: `/image_upload`,
          headers: {
            Authorization: `Bearer tokenifneeded`
          }
        },
        function(err, res) {
          res.on("data", function(response) {
            const parsedResponse = JSON.parse(response.toString("utf8"))
            resolve(parsedResponse.data)
          })
        }
      )
    })
  }
```

**Comment by rodrigoruiz** [comment]
> I have express server.
> I get file from browser in my express server
> How add this file to form-data that send file to another server?
> 
> express
> 
> ```
> const FormData = require('form-data')
> 
> in route
> 
> const files = req.files (files from browser)
> const form = new FormData()
> 
> files.forEach((file)=>{
>         // in file i have only buffer info
>         // if i send file not file.buffer i get error: **source.on is not a function**.
>         // therefore i send file.buffer
>         form.append('file', file.buffer, file.originalname)
>     });
> 
> axios.post('anotherserver', form, {headers: 'multipart/form-data'})
> ```
> 
> It right way or not? How i can see my file in formData?
> After request to another server i can see
> 
> ```
> FormData {
>  _overheadLength: 154,
> _valueLength: 4096,
> _valuesToMeasure: [],
> writable: false,
> readable: true,
> dataSize: 0,
> maxDataSize: 2097152,
> pauseStreams: true,
> _released: true,
> _streams: [],
> _currentStream: null,
> _boundary: '--------------------------181997849912733363348082',
> _events: {},
> _eventsCount: 0 } },
> ```
> 
> Where is storage my file?

Did you end up finding a solution?

**Comment by rodrigoruiz** [comment]
> Here is what I used
> 
> ```
> sendToExternalService(file) {
>     return new Promise(function(resolve, reject) {
>       let formData = new FormData()
>       formData.append("image", file.buffer, file.originalname)
>       formData.submit(
>         {
>           protocol: "https:",
>           host: `example.com`,
>           path: `/image_upload`,
>           headers: {
>             Authorization: `Bearer tokenifneeded`
>           }
>         },
>         function(err, res) {
>           res.on("data", function(response) {
>             const parsedResponse = JSON.parse(response.toString("utf8"))
>             resolve(parsedResponse.data)
>           })
>         }
>       )
>     })
>   }
> ```
> 
> Then you can call `await sendToExternalService(file)`

File comes from `request.files['some_name_from_front_end']` and does not have a `buffer` property.

**Comment by Jeremiah-Ropo** [comment]
Please have you find a solution to it... Or can anyone navigate where I could find the solution? I'm having a very tough time with this thing also, here.
